src/node.cc

```diff
@@ -967,19 +967,21 @@ Local<Value> UVException(Isolate* isolate,
 // Look up environment variable unless running as setuid root.
 bool SafeGetenv(const char* key, std::string* text) {
 #ifndef _WIN32
-  if (getuid() != geteuid() || getgid() != getegid()) {
-    text->clear();
-    return false;
-  }
+  if (getuid() != geteuid() || getgid() != getegid())
+    goto fail;
 #endif
-  if (linux_at_secure) {
-    text->clear();
-    return false;
-  }
+
+#if defined(__linux__)
```

Comment:
I might have misunderstood the [code review](https://github.com/nodejs/node/pull/12548#discussion_r117185327) but it was intentional to not have the macro guard (\_\_linux\_\_) here. Perhaps the variable should also be guarded in [node_main.cc](https://github.com/nodejs/node/blob/ab28e0a86a82eb3f7363b980fcb91261d7e7a2a0/src/node_main.cc#L84-L86) and [node.cc](https://github.com/nodejs/node/blob/ab28e0a86a82eb3f7363b980fcb91261d7e7a2a0/src/node.cc#L233)?